File: tailoring-online-services/user-management-service/src/main/java/com/user_management_service/dto/UserDto.java
package com.user_management_service.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.user_management_service.enums.*;
import lombok.*;

import java.io.Serializable;
import java.sql.Date;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class UserDto implements Serializable {
    private String username;
    @JsonProperty(access = JsonProperty.Access.WRITE_ONLY)
    private String password;
    private String email;
    private String firstName;
    private String lastName;
    private String phoneNumber;
    private String profilePicture;
    private Date dateOfBirth;
    private Date lastLogin;
    private Status status;
    private LanguagePreference languagePreference;
    private Gender gender;
    private NotificationPreference notificationPreference;
    private Boolean emailVerified;
    private Boolean phoneVerified;
    private Boolean OAuth2;
    private Boolean is2FAuth;
    private Boolean hasFingerprint;
    private Boolean hasFaceId;
    private Boolean isVerified;
}

File: tailoring-online-interfaces/src/app/core/dtos/user.dto.ts
import { Gender } from "../enums/gender.enum";
import { LanguagePreference } from "../enums/language-preference.enum";
import { NotificationPreference } from "../enums/notification-preference.enum";
import { Status } from "../enums/status.enum";

export interface UserDto {
  username: string;
  password: string;
  email: string;
  firstName: string;
  lastName: string;
  phoneNumber: string;
  profilePicture?: string;
  dateOfBirth: Date;
  lastLogin: Date;
  status: Status;
  languagePreference: LanguagePreference;
  gender: Gender;
  notificationPreference: NotificationPreference;
  emailVerified: boolean;
  phoneVerified: boolean;
  OAuth2: boolean;
  is2FAuth: boolean;
  hasFingerprint: boolean;
  hasFaceId: boolean;
  isVerified: boolean;
}
